emcc.py

```diff
@@ -2598,7 +2598,10 @@ def modularize():
 %(src)s
 
   return %(EXPORT_NAME)s;
-}%(instantiate)s;
+};
+%(EXPORT_NAME)s = %(EXPORT_NAME)s.bind({
+  _currentScript: typeof document !== 'undefined' ? document.currentScript : undefined
+})%(instantiate)s;
```

Comment:
What do you think about leaving this up to the user, instead? We could document that `document.currentScript` is used, and so if the page will modify that, the user should implement `locateFile` in order to get things working properly.

Alternatively, we could make more of an effort here: write up tests that ensure the right behavior, and document what cases exactly we handle. Basically, it's a "best-effort" without clear guidelines and tests that I'd prefer to avoid.

Reply:
Well, my idea was to make it just work by default and it does. I'd like to avoid user to require doing anything special just to get this basic thing working, re-implementing `locateFile` would be a big pain, especially since it would be necessary to get script's `document.currentScript` from the outside.

I'm more inclined to write more tests here rather than dealing with it every time I use Emscripten.